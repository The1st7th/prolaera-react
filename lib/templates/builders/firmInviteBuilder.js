'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactHtmlEmail = require('react-html-email');

var _button = require('../../components/button');

var _button2 = _interopRequireDefault(_button);

var _footer = require('../../components/footer');

var _footer2 = _interopRequireDefault(_footer);

var _genericInfo = require('../../components/generic/genericInfo');

var _genericInfo2 = _interopRequireDefault(_genericInfo);

var _header = require('../../components/header');

var _header2 = _interopRequireDefault(_header);

var _subHeader = require('../../components/subHeader');

var _subHeader2 = _interopRequireDefault(_subHeader);

var _templateCSS = require('../templateCSS.js');

var _templateCSS2 = _interopRequireDefault(_templateCSS);

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

function _asyncToGenerator(fn) {
  return function() {
    var gen = fn.apply(this, arguments);
    return new Promise(function(resolve, reject) {
      function step(key, arg) {
        try {
          var info = gen[key](arg);
          var value = info.value;
        } catch (error) {
          reject(error);
          return;
        }
        if (info.done) {
          resolve(value);
        } else {
          return Promise.resolve(value).then(
            function(value) {
              step('next', value);
            },
            function(err) {
              step('throw', err);
            }
          );
        }
      }
      return step('next');
    });
  };
}

var firmInviteBuilder = (function() {
  var _ref = _asyncToGenerator(
    /*#__PURE__*/ regeneratorRuntime.mark(function _callee(completeUser, completeCompany, imageUrl) {
      var headerText, bodyText, footerText, Header;
      return regeneratorRuntime.wrap(
        function _callee$(_context) {
          while (1) {
            switch ((_context.prev = _context.next)) {
              case 0:
                _context.prev = 0;
                headerText = '';
                bodyText = _react2.default.createElement(
                  'div',
                  null,
                  _react2.default.createElement(
                    'p',
                    null,
                    'Your account is waiting for you and there are only a couple quick steps to complete.'
                  ),
                  _react2.default.createElement(
                    'ol',
                    null,
                    _react2.default.createElement('li', null, 'Click the button below to register your account.'),
                    _react2.default.createElement(
                      'li',
                      null,
                      'On the registration page simply enter a password for your new account and click "Register".'
                    )
                  )
                );
                footerText = _react2.default.createElement(
                  'div',
                  null,
                  _react2.default.createElement(
                    'p',
                    null,
                    "That's it! You're ready to explore the new tools and resources you have access to."
                  ),
                  _react2.default.createElement(_button2.default, {
                    text: 'Create Your Account',
                    link: 'https://app.prolaera.com/#/registration?email=${' + completeUser.email + '}' + "'"
                  })
                );
                _context.next = 6;
                return (0, _header2.default)(imageUrl);

              case 6:
                Header = _context.sent;
                return _context.abrupt(
                  'return',
                  (0, _reactHtmlEmail.renderEmail)(
                    _react2.default.createElement(
                      _reactHtmlEmail.Email,
                      {
                        title: "You've been added to " + ('' + completeCompany.company_name) + 'on Prolaera',
                        headCSS: _templateCSS2.default
                      },
                      _react2.default.createElement(Header, {
                        text: 'Create Account',
                        link: 'https://app.prolaera.com/#/registration?email=${' + completeUser.email + '}' + "'"
                      }),
                      _react2.default.createElement(_subHeader2.default, {
                        subText: "You've been",
                        header: 'Added',
                        message:
                          'You have been added to ' + ('' + completeCompany.company_name) + "'s team on Prolaera."
                      }),
                      _react2.default.createElement(_genericInfo2.default, {
                        header: headerText,
                        body: bodyText,
                        footer: footerText
                      }),
                      _react2.default.createElement(_footer2.default, null)
                    )
                  )
                );

              case 10:
                _context.prev = 10;
                _context.t0 = _context['catch'](0);
                throw _context.t0;

              case 13:
              case 'end':
                return _context.stop();
            }
          }
        },
        _callee,
        undefined,
        [[0, 10]]
      );
    })
  );

  return function firmInviteBuilder(_x, _x2, _x3) {
    return _ref.apply(this, arguments);
  };
})();

exports.default = firmInviteBuilder;
