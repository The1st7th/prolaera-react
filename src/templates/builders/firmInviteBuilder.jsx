import React from 'react';
import { Email, renderEmail } from 'react-html-email';
import Button from '../../components/button';
import Footer from '../../components/footer';
import GenericInfo from '../../components/generic/genericInfo';
import builderHeader from '../../components/header';
import SubHeader from '../../components/subHeader';
import css from '../templateCSS';

const firmInviteBuilder = async (completeUser, completeCompany, imageUrl) => {
  try {
    const headerText = '';
    const bodyText = (
      <div>
        <p>Your account is waiting for you and there are only a couple quick steps to complete.</p>
        <ol>
          <li>Click the button below to register your account.</li>
          <li>On the registration page simply enter a password for your new account and click "Register".</li>
        </ol>
      </div>
    );
    const footerText = (
      <div>
        <p>That's it! You're ready to explore the new tools and resources you have access to.</p>
        <div
          style={{
            backgroundColor: '#72C02C',
            borderRadius: '3px',
            width: '200px',
            height: '50px',
            margin: 'auto',
            marginBottom: '3px',
            textAlign: 'middle'
          }}
        >
          <Button
            text={'Create Your Account'}
            link={'https://app.prolaera.com/#/registration?email=${' + completeUser.email + '}' + "'"}
          />
        </div>
      </div>
    );
    const Header = await builderHeader(imageUrl);
    return renderEmail(
      <Email title={"You've been added to " + `${completeCompany.company_name}` + 'on Prolaera'} headCSS={css}>
        <Header
          text={'Create Account'}
          link={'https://app.prolaera.com/#/registration?email=${' + completeUser.email + '}' + "'"}
        />
        <SubHeader
          subText={"You've been"}
          header={'Added'}
          message={'You have been added to ' + `${completeCompany.company_name}` + "'s team on Prolaera."}
        />
        <GenericInfo header={headerText} body={bodyText} footer={footerText} />
        <Footer />
      </Email>
    );
  } catch (error) {
    throw error;
  }
};

export default firmInviteBuilder;
