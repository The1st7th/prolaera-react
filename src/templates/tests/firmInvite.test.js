import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/button';
import GenericInfo from '../../components/generic/genericInfo';
import writeFile from '../../helpers/writeFileHelper';
import firmInviteBuilder from '../builders/FirmInviteBuilder.jsx';

Enzyme.configure({ adapter: new Adapter() });

const user = { email: 'test@email.com' };
const company = { company_name: 'Test CPA Firm' };
const headerText = <p>You've been added to {company.company_name} on Prolaera</p>;
const bodyText = (
  <div>
    <p>Your account is waiting for you and there are only a couple quick steps to complete.</p>
    <ol>
      <li>Click the button below to register your account.</li>
      <li>On the registration page simply enter a password for your new account and click "Register".</li>
    </ol>
  </div>
);
const footerText = (
  <div>
    <p>You're ready to explore the new tools and resources you have access to.</p>
    <Button text={'Create Your Account'} link={'https://app.prolaera.com/#/users/' + user.email} />
  </div>
);

describe('Firm Invite component', () => {
  it('creates and checks snapshot of FirmInvite component html', async () => {
    let inviteHtml = renderer.create(
      <GenericInfo header={headerText} body={bodyText} footer={footerText} email={user.email} />
    );
    let inviteJson = inviteHtml.toJSON();
    expect(inviteJson).toMatchSnapshot();
  });
});

describe('create html test file', () => {
  it('writes an html file to firmInviteTest.html', async () => {
    const firmInviteEmail = await firmInviteBuilder(
      user,
      company,
      'https://assets.prolaera.com/prolaeraLogo_fullText.png'
    );
    const saved = await writeFile(firmInviteEmail, 'firmInviteTest.html');
    expect(saved).toEqual(true);
  });
});
