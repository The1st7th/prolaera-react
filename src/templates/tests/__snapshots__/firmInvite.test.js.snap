// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Firm Invite component creates and checks snapshot of FirmInvite component html 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": "5px",
      "display": "block",
      "fontFamily": "Arial, sans-serif",
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "500px",
      "padding": "20px",
      "textAlign": "left",
    }
  }
>
  <table
    align="center"
    border="0"
    cellPadding={0}
    cellSpacing={0}
    valign="top"
    width="100%"
  >
    <tbody>
      <tr>
        <td>
          <div>
            <p>
              You've been added to 
              Test CPA Firm
               on Prolaera
            </p>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            <div>
              <p>
                Your account is waiting for you and there are only a couple quick steps to complete.
              </p>
              <ol>
                <li>
                  Click the button below to register your account.
                </li>
                <li>
                  On the registration page simply enter a password for your new account and click "Register".
                </li>
              </ol>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            <div>
              <p>
                That's it! You're ready to explore the new tools and resources you have access to.
              </p>
              <div
                align="center"
                className="customButton"
                style={
                  Object {
                    "maxWidth": "100%",
                    "padding": "20px 0px 20px 0px",
                    "textAlign": "center",
                  }
                }
              >
                <a
                  href="https://app.prolaera.com/#/users/test@email.com"
                  style={
                    Object {
                      "borderRadius": "3px",
                      "color": "white",
                      "padding": "14px 14px 14px 14px",
                      "textDecoration": "none",
                      "width": "200px",
                    }
                  }
                >
                  Create Your Account
                </a>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
